hail/src/main/resources/include/hail/Utils.h

```diff
@@ -129,4 +133,19 @@ class ArrayLoadImpl : public BaseArrayImpl<elem_required, elem_size, elem_align>
   }
 };
 
+struct HailFatalError: public std::exception {
+  private:
+    static constexpr int max_error_len = 512;
+    char error_msg[max_error_len];
+  public:
+    virtual const char * what() const throw() { return error_msg; }
+    virtual ~HailFatalError() throw() {};
+    explicit HailFatalError(const char * fmtstring, ...) : std::exception() {
```

Comment:
Richard put the error number in the message to make it easy to grep the source code for the error.  That means each source line that produces an error should have its own unique id.  Should the exception constructor take the id so we can keep tracking that?

Reply:
hmm... When I'm looking for the error in the (non-generated) source code, I typically just search for the error string, which I feel like should aim to be descriptive and unique enough to find it. (On top of that, I find it hard to keep track of what numbers have been used for error codes so far.)

I think there's something to be said for having unique error IDs in the generated source code, so that we can figure out where in the generated source it comes from if e.g. there's two separate ArrayRefs---does that seem appropriate?